modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1757,10 +1757,10 @@ def fold(self, axis, func):
         )
         return self.__constructor__(
             new_partitions,
-            self.index,
-            self.columns,
-            self.row_lengths,
-            self.column_widths,
+            self._index_cache,
+            self._columns_cache,
+            self._row_lengths_cache,
+            self._column_widths_cache,
```

Comment:
This should be reverted since `map_axis_partitions` doesn't match the map pattern behavior (`map_partitions`).

Reply:
Sorry, this is `Fold` operator that states `The data shape is not changed (length and width of the table).`.